### DISCUSSION TITLE
Using the RESTful API to programmatically trigger the same workflow as the "Upload a CSV" option in the "Data" menu

### DISCUSSION THREAD

**Discussion by ericdrosas87** [apache/superset#23270] [Q&A / Help]
For starters, I have already seen [this discussion thread](https://github.com/apache/superset/discussions/14269), but it does not answer this question. 

I didn't see anything in the API documentation that would allow me to POST to an endpoint and send a CSV as a payload to the API. Is this "Upload a CSV" functionality not exposed as a part of the RESTful API? If not, is there a plugin or module that adds this functionality to the API?

**Comment by gautierrog** [comment]
Hello, we were wondering the same thing here. Have you find anything?

**Comment by ericdrosas87** [comment]
> Hello, we were wondering the same thing here. Have you find anything?

I ended up taking @MPalix86's [repo](https://github.com/MPalix86/programmatically_upload_csv_in_superset) and turning it into a [Google Cloud Function API](https://github.com/lsst-epo/superset-sideloader). It wouldn't take much work to extract the core functionality from the Google Cloud Function framework for your use case though?

**Reply by MPalix86** [reply]
In my configuration, I added the script to an Airflow environment on my server. I then trigger the Airflow DAG each time CSV files are uploaded to a specified folder using a Python library called watchdog to monitor the directory. You can also set up periodic checks for files using Airflow itself!
